src/L2VoteAggregator.sol

```diff
@@ -3,16 +3,20 @@ pragma solidity ^0.8.20;
 
 import {ERC20Votes} from "openzeppelin/token/ERC20/extensions/ERC20Votes.sol";
 import {SafeCast} from "openzeppelin/utils/math/SafeCast.sol";
+import {EIP712} from "openzeppelin/utils/cryptography/EIP712.sol";
+import {ECDSA} from "openzeppelin/utils/cryptography/ECDSA.sol";
 
 import {L2GovernorMetadata} from "src/WormholeL2GovernorMetadata.sol";
 import {IL1Block} from "src/interfaces/IL1Block.sol";
 
 /// @notice A contract to collect votes on L2 to be bridged to L1.
-abstract contract L2VoteAggregator {
+abstract contract L2VoteAggregator is EIP712 {
   /// @notice The number of blocks before L2 voting closes. We close voting 1200 blocks before the
   /// end of the proposal to cast the vote.
   uint32 public constant CAST_VOTE_WINDOW = 1200;
 
+  bytes32 public constant BALLOT_TYPEHASH = keccak256("Ballot(uint256 proposalId,uint8 support)");
```

Comment:
Also, I want to highlight this typehash to make sure it is correct.

Reply:
Seems to match with the one defined [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/Governor.sol#L30-L31), except for the nonce. We don't have a nonce, because as of now, we only let each L2 voter vote once. While we might want to consider that assumption in the future, it's fine for now.

Reply:
I based it on this version of the Governor which is also missing voter. The Tally docs have a signature that differs from the current Governor signature for `castVoteBySig`

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/cd48b3eab380254b08d7893a5a7bf568a33c5259/contracts/governance/Governor.sol#L31

Reply:
Yep! I think this is fine for now. I think Tally hasn't updated yet to account for the nonce. It's a new addition by OZ, and came in direct response to our finding in Flexible Voting.

Reply:
So I think this is good to go 👍